crates/stages/src/pipeline/ctrl.rs

```diff
@@ -32,4 +32,13 @@ impl ControlFlow {
     pub fn is_unwind(&self) -> bool {
         matches!(self, ControlFlow::Unwind { .. })
     }
+
+    /// Returns the pipeline progress, if the state is not `Unwind`.
+    pub fn progress(&self) -> Option<BlockNumber> {
```

Comment:
I think we should consider renaming this and getting rid of the word "progress" because there is some ambiguity here. 

what does this really mean?
number of blocks since then last invokation? or current height after pipeline run?

but for now this is ok, because this follows the controlflow variant

cc @shekhirin

Reply:
> or current height after pipeline run?

Ideally this is the correct definition, or at it's least what we need for continuous sync

Reply:
agree, I feel like progress has multiple definitions in the codebase and ideally it'd be nice to separate it into something like `checkpoint` (we use it to continue execution or unwind on next iteration), `block_height` (highest block number reached by a stage) and actual `progress` (stage-specific progress or block number)